### PR TITLE
Feature: query-and-write delete command

### PR DESCRIPTION
**Pull request by imforster**
#39 
Test cases: 88
Docs: https://www.mongodb.com/docs/v8.2/reference/command/delete/

Adds compatibility test coverage for the query-and-write delete command, validating its full surface area - from basic document deletion and limit semantics to BSON type distinction, null/missing field matching, ordered/unordered error handling, $expr filter support, hint/let/collation option wiring, and argument type rejection - across document types and edge cases.

88 test cases across 7 files following project test guidelines.

Also renames the query-and-write directory tree to use underscores (query_and_write) for valid Python package naming, and renames read-concern and write-concern subdirectories accordingly.

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_error_cases.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    assertResult(
+        result,
+        expected=test.build_expected(ctx),
+        error_code=test.error_code,
+        msg=test.msg,
+        raw_res=True,
+    )
+
+
+def test_delete_deletes_non_array_object_fails(collection):
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_error_cases.py]: `assertExceptionType(result, TypeError, ...)` here (and at line 219) checks a client-side PyMongo `TypeError` raised before the command leaves the driver — these test the driver, not the server. Either remove these or move them to a clearly-labeled client-side rejection section with a docstring explaining why they don't go through `assertResult(error_code=...)` like the rest of the file.

File: documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_bulk_operations.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        },
+    )
+    assertSuccessPartial(
+        result,
+        {"ok": 1.0, "n": 2, "writeErrors": [{"index": 1, "code": BAD_VALUE_ERROR}]},
+        msg="delete ordered:false should continue past errors",
+    )
+
+
+def test_delete_ordered_true_error_at_first(collection):
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_bulk_operations.py:153]: `assertSuccessPartial` extracts only the keys present in `expected`. Asserting `writeErrors: [{"index": 0, "code": ...}]` will pass even if `writeErrors` actually contains multiple entries — so this test won't fail if subsequent statements somehow also produced errors. To actually pin down "stops immediately on first error" (and similarly "continues but reports exactly one error" at line 158), assert the array length explicitly (e.g. via `Len(1)` or a separate length check).

File: documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_bulk_operations.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+            "deletes": [
+                {"q": {"status": "D"}, "limit": 0},
+                {"q": {"status": "A"}, "limit": 1},
+            ],
+        },
+        expected={"ok": 1.0, "n": 4},
+        msg="delete should handle mixed limit:0 and limit:1 in bulk",
+    ),
+    CommandTestCase(
+        "duplicate_target_ordered",
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/core/query_and_write/commands/delete/test_delete_bulk_operations.py]: The case ID is `duplicate_target_ordered` but the command doesn't set `ordered`. Either rename to drop `_ordered` or add `"ordered": True` to make the ID match the behavior under test.

**Review by eerxuan — APPROVED** [review]

**Review by eerxuan — APPROVED** [review]
